server/src/main/java/org/apache/iotdb/db/sync/externalpipe/ExtPipePluginManager.java

```diff
@@ -176,46 +194,69 @@ private void monitorPipeData() {
       return;
     }
 
-    while (true) {
-      List<PipeData> pipeDataList = tsFilePipe.pull(Long.MAX_VALUE);
-      if ((pipeDataList != null)
-          && (!pipeDataList.isEmpty())
-          && (pipeDataList.get(pipeDataList.size() - 1).getSerialNumber()
-              > lastPipeDataSerialNumber)) {
-        for (PipeData pipeData : pipeDataList) {
-          long pipeDataSerialNumber = pipeData.getSerialNumber();
-          if (pipeDataSerialNumber <= lastPipeDataSerialNumber) {
-            continue;
-          }
-          lastPipeDataSerialNumber = pipeData.getSerialNumber();
+    while (alive) {
+      try {
+        // == pull Pipe src data and insert them to pipeOpManager
+        List<PipeData> pipeDataList = tsFilePipe.pull(Long.MAX_VALUE);
+        if ((pipeDataList != null)
+            && (!pipeDataList.isEmpty())
+            && (pipeDataList.get(pipeDataList.size() - 1).getSerialNumber()
+                > lastPipeDataSerialNumber)) {
+          for (PipeData pipeData : pipeDataList) {
+            long pipeDataSerialNumber = pipeData.getSerialNumber();
+            if (pipeDataSerialNumber <= lastPipeDataSerialNumber) {
+              continue;
+            }
 
-          // extract the Tsfile PipeData
-          if (pipeData instanceof TsFilePipeData) {
-            TsFilePipeData tsFilePipeData = (TsFilePipeData) pipeData;
+            // == extract the Tsfile PipeData
+            if (pipeData instanceof TsFilePipeData) {
+              TsFilePipeData tsFilePipeData = (TsFilePipeData) pipeData;
+
+              String sgName = tsFilePipeData.getStorageGroupName();
```

Comment:
Delete the storageGroupName field in TsFilePipe, you can use `StorageEngine.getSgByEngineFile()` instead

Reply:
@Cpaulyz  Thank your comments.
Here, I prefer to keep current design (using storageGroupName ). Because I feel curent desgin has higher effeciency.